src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -1211,46 +1210,14 @@ class AddItemsPage extends Component {
     };
     if (payload.lineItems.length) {
       return apiClient.post(updateItemsUrl, payload)
-        .then((resp) => {
-          const { lineItems } = resp.data.data;
-          const lineItemsBackendData = _.map(
-            lineItems,
-            val => ({
-              ...val,
-              product: {
-                ...val.product,
-                label: `${val.productCode} ${val.product.name}`,
-              },
-              quantityOnHand: _.find(
-                this.state.values.lineItems,
-                lineItem => lineItem.sortOrder === val.sortOrder,
-              ).quantityOnHand,
-              monthlyDemand: _.find(
-                this.state.values.lineItems,
-                lineItem => lineItem.sortOrder === val.sortOrder,
-              ).monthlyDemand,
-              demandPerReplenishmentPeriod: _.find(
-                this.state.values.lineItems,
-                lineItem => lineItem.sortOrder === val.sortOrder,
-              ).demandPerReplenishmentPeriod,
-              quantityAvailable: _.find(
-                this.state.values.lineItems,
-                lineItem => lineItem.sortOrder === val.sortOrder,
-              ).quantityAvailable,
-            }),
-          );
-
-          this.setState({
-            values:
-              { ...this.state.values, lineItems: lineItemsBackendData },
-            totalCount: lineItems.length,
-            currentLineItems: lineItemsBackendData,
-          });
-        })
         .then(() => this.fetchAddItemsPageData())
+        .then(() => this.setState({
```

Comment:
setting empty lists for  `currentLineItems: []` and  `values: { ...this.state.values, lineItems: [] }` is necessary, because `fetchLineItems()` method concatenates previouse lineItems with the new fetched ones

Reply:
Move `this.fetchAddItemsPageData()` from a separate `then` to the setState's callback